source/table_skel.c

```diff
@@ -0,0 +1,112 @@
+#include "table_skel.h"
+#include "data.h"
+#include "sdmessage.pb-c.h"
+
+/* Inicia o skeleton da tabela.
+ * O main() do servidor deve chamar esta função antes de poder usar a
+ * função invoke(). O parâmetro n_lists define o número de listas a
+ * serem usadas pela tabela mantida no servidor.
+ * Retorna a tabela criada ou NULL em caso de erro.
+ */
+struct table_t *table_skel_init(int n_lists){
+    return table_create(n_lists);
+}
+
+/* Liberta toda a memória ocupada pela tabela e todos os recursos 
+ * e outros recursos usados pelo skeleton.
+ * Retorna 0 (OK) ou -1 em caso de erro.
+ */
+int table_skel_destroy(struct table_t *table){
+    return table_destroy(table);
+}
+
+/* Executa na tabela table a operação indicada pelo opcode contido em msg 
+ * e utiliza a mesma estrutura MessageT para devolver o resultado.
+ * Retorna 0 (OK) ou -1 em caso de erro.
+*/
+int invoke(MessageT *msg, struct table_t *table){
+    int res;
+    switch (msg->opcode){
+        case MESSAGE_T__OPCODE__OP_BAD:
```

Comment:
Unfortunately, you have to wrap the code for the case in curly braces
```suggestion
        case MESSAGE_T__OPCODE__OP_BAD: {
            // Code here
        }
```

Reply:
![image](https://github.com/enderesting/sd-projeto-2/assets/1679672/c0007d18-efc7-4327-bc49-00b25964deeb)

Reply:
oh weird, i don't remember switch case needing curly brackets...

Reply:
nvm, if u declare variables in it it does need them